xdsl/irdl/irdl.py

```diff
@@ -712,28 +718,47 @@ class AttributeDef:
     constr: AttrConstraint
     """The attribute constraint."""
 
-    attr_name: str | None = None
```

Comment:
I am confused about this change. Why is it necessary? Also, does this break our generic attribute format? Or does this now just live somewhere else?

Reply:
We had kind of duplicated data, so we could grab this from somewhere else.
It doesn't change anything otherwise, it lives in another structure already.